application/models/Survey.php

```diff
@@ -1824,8 +1825,62 @@ public static function getSurveysWithTokenTable()
         return $surveys;
     }
 
+    /**
+     * Add needed language if needed in related SurveyLanguageSetting
+     * Remove uneeded language if needed in related SurveyLanguageSetting
+     * @return void
+     */
+    public function fixSurveyLanguageConsistency()
```

Comment:
Maybe put it in a service class...?

Reply:
/me tired …

Reply:
I know, sorry. :( But logic shouldn't really be in model classes. I can fix it if you want.

Reply:
> logic shouldn't really be in model class

Unsure on this point.

I can move it it's not the issue, but don't ask me to move common_helper function `fixLanguageConsistency` in class too … (because i think it's more important …)

See : 

>  Dev: can be improved but need a start point

Reply:
> logic shouldn't really be in model class

It's about leaving the "fat model"-paradigm behind, to increase testability. (Model = ActiveRecord, in this case.)

I wouldn't ask you to refactor something you didn't touch... :| I just want to avoid increasing the technical debt.